### DISCUSSION TITLE
Add header in get capability request

### DISCUSSION THREAD

**Discussion by samareshdasglobal** [TerriaJS/terriajs#7065] [General]
Hello,
I am working on terriMap, We have geoserver which is deployed in aws through API gateway. API gateway needs authorization header in each request. How can i add header in geosever request like getcapability, titles request

**Comment by samareshdasglobal** [comment]
I am able to manage add header during request for get capabiliy. I need to change in this files lib/Core/loadWithXhr.js and lib/Core/loadXML.js in terrijs project like 
if(resource.headers){
 resource.headers['Mytokon']='this is my token ';
  }else{
    resource.headers={'Mytokon':'this is my token';};
  }
  I am still struggling to find out the place to add header when request for wms tiles.

**Comment by nf-s** [comment]
Hi @samareshdasglobal 

TerriaJS currently doesn't have a way to add credentials to requests - TerriaJS-server does have the `proxyAuth` function which allows credentials to be added to requests without revealing them to users

For example
- https://github.com/TerriaJS/terriajs-server/blob/master/proxyauth.json.example

We have a ticket here tracking this issue https://github.com/TerriaJS/terriajs/issues/6704
